lib/features/reps_core/reps_core.install

```diff
@@ -389,3 +389,81 @@ function reps_core_update_7012() {
     $source->startImport();
   }
 }
+
+/**
+ * Forcing reerting reps_core to apply upgrade 7014.
+ */
+function reps_core_update_7013() {
+  features_revert(
+    array(
+      'reps_core' => array(
+        'field_base',
+        'field_instance',
+      ),
+    )
+  );
+}
+
+/**
+ * Add value for existing entities delete unsused rightcol_position field.
+ */
+function reps_core_update_7014() {
+  // Set default value for field_reps_display_region in existing bean blocks.
+  $field_name = 'field_reps_display_region';
+  $entity_type = 'bean';
+  $entities = entity_load($entity_type);
+  $field_instance = field_info_instance($entity_type, $field_name, 'reps_core_sb_right_image_link');
+  $default_value = $field_instance['default_value'][0]['value'];
+  db_update('field_data_field_reps_rightcol_weight')
+    ->fields(array(
+      'language' => LANGUAGE_NONE,
+    ))->execute();
+
+  foreach ($entities as $entity) {
+    if ($entity->type != 'webtools' or $entity->delta == 'bean-twitter') {
+      if (empty($entity->{$field_name})) {
+
+        // Force specific blocks in right columns.
+        $entity->{$field_name}[LANGUAGE_NONE][0]['value'] = $default_value;
+
+        // Twitter feed case.
+        if ($entity->delta == 'bean-twitter') {
+          // Set weight to 0 for twitter bean block.
+          $entity->field_reps_rightcol_weight[LANGUAGE_NONE][0]['value'] = 3;
+          // Get frontpage ID.
+          $site_frontpage = str_replace('node/', '', drupal_get_normal_path(variable_get('site_frontpage')));
+          foreach (language_list() as $language) {
+            $entity->field_reps_core_node_reference[$language->language][0]['target_id'] = $site_frontpage;
+          }
+        }
+
+        // Other blocks weight has to be set.
+        else {
+          // Get positioning base on old field value.
+          $old_position_value = array_values($entity->field_reps_rightcol_position)[0][0]['value'];
+          // Reset old values.
+          $entity->field_reps_rightcol_weight = array();
+          if ($old_position_value == 1) {
+            $entity->field_reps_rightcol_weight[LANGUAGE_NONE][0]['value'] = 1;
+          }
+          else {
+            $entity->field_reps_rightcol_weight[LANGUAGE_NONE][0]['value'] = 5;
+          }
+        }
```

Comment:
Here you could have simply written:
         `if ($old_position_value != 1) {
            $entity->field_reps_rightcol_weight[LANGUAGE_NONE][0]['value'] = 5;
          }`

Reply:
I'm not sure this is correct. Because the old position is used to determine the value new of the field_reps_rightcol_weight. In your proposition there is no way to set the weight to 1. Only 5, or the old value are possible.